### DISCUSSION TITLE
Failed to upgrade Rabbitmq from 3.10.8 to 3.12.3

### DISCUSSION THREAD

**Discussion by yaniv1988** [docker-library/rabbitmq#662] [Q&A]
We are having a Rabbitmq 3.10.8 cluster setup and now we plan to upgrade to Rabbitmq 3.12.3 using rabbitmq:3.12.3-management-alpine docker image. 

Post upgrade, the node fails to start with the error **"Feature flags: `classic_queue_type_delivery_support`: required feature flag not enabled! It must be enabled before upgrading RabbitMQ."**

I am aware that direct upgrade is not possible as mentioned in https://www.rabbitmq.com/feature-flags.html and we have to do an intermediate upgrade to 3.11 and manually enable the necessary feature flags and then upgrade to 3.12.3

Even if I do intermediate upgrade to 3.11 and enable feature flags in Management UI, it is enabling it on one particular node and not on all the nodes in the cluster. As a result, next upgrade to 3.12 fails with same error.

Please suggest an easier way to do the upgrade.

**Comment by lukebakken** [comment]
You must ensure _all_ feature flags are enabled on _all_ nodes prior to any upgrade. Please do that before upgrading to 3.11.

If you still have errors, you have to provide an exact sequence of steps to reproduce the issue. We don't have time to guess what commands you are running.

Finally, this discussion probably should have been started here - https://github.com/rabbitmq/rabbitmq-server/discussions

**Comment by noel-vincent-cde** [comment]
The following solution worked for me.

Current version I was using was 3.11.0 and I was planning to upgrade to 3.12.0
I was getting the same error.

The feature flag "classic_queue_type_delivery_support" is not available in the rabbitMQ version 3.11.0.

This is available in version 3.11.18.

So if you need to upgrade rabbitMQ from 3.11.0 to 3.12.x, you need to first upgrade to 3.11.18 and enable the flag and then upgrade to 3.12.x

I hope this is clear.

**Comment by lukebakken** [comment]
If the documentation is not clear enough, please let us know:

https://www.rabbitmq.com/docs/upgrade

**Reply by ashleykleynhans** [reply]
@lukebakken the documentation is not clear enough, I had to use @NoelVincent 's solution above.

**Reply by michaelklishin** [reply]
@ashleykleynhans there is a table with version upgradeability that does mention specifically `3.11.18` https://www.rabbitmq.com/docs/upgrade#rabbitmq-version-upgradability

**Reply by michaelklishin** [reply]
I have updated the guide on upgrades like so: https://github.com/rabbitmq/rabbitmq-website/commit/2bf04899b3a7178565ffec8d197cf4f413505faf

It now explicitly mentions that for every intermediary (or any, really) upgrade, the expectation is that the user upgrades to the most recent patch release in the series.
